sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -419,3 +419,30 @@ case class Greatest(children: Seq[Expression]) extends Expression {
     """
   }
 }
+
+/** Operator that drops a row when it contains any nulls. */
+case class DropAnyNull(child: Expression) extends UnaryExpression {
```

Comment:
Yeah, this is mostly so that we get nice errors if another developer misuses the API in the future (i.e. before the class cast at runtime).  Should be pretty simple.